src/core/utils/x25519.ts

```diff
@@ -0,0 +1,9 @@
+import { x25519 } from "@noble/curves/ed25519";
+
+export function getX25519PrivateKey(): Uint8Array {
+  return x25519.utils.randomPrivateKey();
```

Comment:
I suspect this needs the bit manipulation of the top and bottom bits: 
       k[0] &= 248
       k[31] &= 127
       k[31] |= 64

Reply:
@KomelT if this is working now please resolve this comment